README.md

```diff
@@ -6,66 +6,93 @@ An Ansible role for bootloader and kernel command line management.
+* `state` - `state` (default) or `absent`. The value `absent` means to remove a setting with `name` name - name must be provided.
+* `previous` - Optional - the only value is `replaced` - this is used to specify that the previous values in a group should be replaced with the given values.
 
-```yaml
-bootloader_foo: "oof"
-bootloader_bar: "baz"
-```
+Default: `[]`
 
-### Variables Exported by the Role
+Type: `list`
 
-This section is optional.  Some roles may export variables for playbooks to
-use later.  These are analogous to "return values" in Ansible modules.  For
-example, if a role performs some action that will require a system reboot, but
-the user wants to defer the reboot, the role might set a variable like
-`bootloader_reboot_needed: true` that the playbook can use to reboot at a more
-convenient time.
+### bootloader_timeout
 
-Example:
+With this variable, you can customize the loading time of the GRUB bootloader.
 
-`bootloader_reboot_needed` - default `false` - if `true`, this means
-a reboot is needed to apply the changes made by the role
+Default: `5`
 
-## Example Playbook
+Type: `int`
+
+### bootloader_password
+
+With this variable, you can protect boot parameters with a password.
+Boot loader username is always `root`.
+This should come from vault.
+If unset, current configuration is not touched.
+
+Default: `null`
+
+Type: `string`
+
+### bootloader_remove_password
+
+By setting this variable to `true`, you can remove boot loader password.
+
+Default: `false`
+
+Type: `bool`
+
+### bootloader_reboot_ok
+
+If `true`, if the role detects that something was changed that requires a reboot to take effect, the role will reboot the managed host.
 
-Including an example of how to use your role (for instance, with variables
-passed in as parameters) is always nice for users too:
+If `false`, it is up to you to determine when to reboot the managed host.
+
+The role will returns the variable `bootloader_reboot_required` (see below) with a value of `true` to indicate that some change has occurred which needs a reboot to take effect.
+
+Default: `false`
+
+Type: `bool`
+
+### Variables Exported by the Role
+
+The role exports the following variables:
+
+`bootloader_reboot_needed` - default `false` - if `true`, this means a reboot is needed to apply the changes made by the role
+
+## Example Playbook
 
 ```yaml
 - hosts: all
   vars:
-    bootloader_foo: "foo foo!"
-    bootloader_bar: "progress bar"
-
+    bootloader_settings:
+      - name: crashkernel
+        value: auto
+      - name: no_timer_check
+        state: present
+      - name: quiet
+        state: absent
+      - name: debug
+        state: present
+      - name: rd.lvm.lv
+        state: present
+      - name: resume
+    bootloader_timeout: 5
+    bootloader_password: null
+    bootloader_remove_password: false
+    bootloader_reboot_ok: true
   roles:
     - linux-system-roles.bootloader
 ```
```

Comment:
We should have some examples of that show the kernel command line before running the role, and what the command line looks like after running the role will various parameters e.g.
```
before
args="rhgb quiet"

with parameters as above will result in
args="rhgb quiet crashkernel=auto no_timer_check debug rd.lvm.lv resume"

with parameters as above with `previous: replaced` will result in
args="quiet crashkernel=auto no_timer_check debug rd.lvm.lv resume"
```
most people using this role will be familiar with `grubby` other tools that let you edit the kernel command line in a similar format, so having examples like that should help.